### ISSUE TITLE
Instead of an exception traceback, ansible-lint should show a proper error.

### ISSUE THREAD

**Issue by rjeffman** [ansible-community/ansible-lint#1012]
##### Summary

When running ansible-lint on a playbook with syntax-error, it should show that a syntax error exists, and not throw an exception (like TypeError).

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console
ansible --version

ansible 2.9.12
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/rjeffman/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/rjeffman/Developer/ansible-freeipa/.venv/lib64/python3.8/site-packages/ansible
  executable location = /home/rjeffman/Developer/ansible-freeipa/.venv/bin/ansible
  python version = 3.8.5 (default, Jul 31 2020, 00:00:00) [GCC 10.2.1 20200723 (Red Hat 10.2.1-1)]

ansible-lint --version

ansible-lint 4.3.3
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

Fedora 32, Python 3.8.5, runing in a `venv` virtual environment.

##### STEPS TO REPRODUCE

Run `ansible-lint` on the playbook:

```yaml
---
- name: Force exception on ansible-lint
  hosts: localhost

  tasks:
    debug:
      msg: "Note that `tasks` is not entered as a list."
```

##### Desired Behaviour

In the presence or syntax errors, ansible-lint should report them as syntax errors, an not show an unhandled exception.

##### Actual Behaviour

```console
[rjeffman@dantooine ansible-freeipa] (ci_add_yml_linters)$ python3 -m ansiblelint test.yml 
Traceback (most recent call last):
  File "/usr/lib64/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib64/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.8/site-packages/ansiblelint/__main__.py", line 176, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/ansiblelint/__main__.py", line 154, in main
    matches.extend(runner.run())
  File "/usr/local/lib/python3.8/site-packages/ansiblelint/runner.py", line 78, in run
    matches = set(self._emit_matches(files))
  File "/usr/local/lib/python3.8/site-packages/ansiblelint/runner.py", line 103, in _emit_matches
    for child in ansiblelint.utils.find_children(arg, self.playbook_dir):
  File "/usr/local/lib/python3.8/site-packages/ansiblelint/utils.py", line 164, in find_children
    for child in _rebind_match_filename(playbook[0], play_children)(
  File "/usr/local/lib/python3.8/site-packages/ansiblelint/utils.py", line 143, in func_wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/ansiblelint/utils.py", line 217, in play_children
    return delegate_map[k](basedir, k, v, parent_type)
  File "/usr/local/lib/python3.8/site-packages/ansiblelint/utils.py", line 235, in _taskshandlers_children
    children = _get_task_handler_children_for_tasks_or_playbooks(
  File "/usr/local/lib/python3.8/site-packages/ansiblelint/utils.py", line 270, in _get_task_handler_children_for_tasks_or_playbooks
    'path': path_dwim(basedir, task_handler[task_handler_key]),
TypeError: string indices must be integers
```
